poolmgr/api.go

```diff
@@ -135,6 +137,30 @@ func (api *API) getServiceForFunction(m *fission.Metadata) (string, error) {
 		return fsvc.address, nil
 	}
 
+	// Cache miss -- is this first one to request the func?
+	createCh, found := api.fsCreateChan[*m]
```

Comment:
Since the API is called in multiple goroutines, this access to the map is unsafe, as is the one below that adds to the map.